src/main/java/core/basesyntax/service/db/StorageServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.service.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StorageServiceImpl implements StorageService {
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    @Override
+    public void updateBalance(String fruit, int amount) {
+        storage.put(fruit, amount);
+    }
```

Comment:
The updateBalance method is not safe. It can overwrite an existing value in the storage. It would be better to check if the fruit exists in the storage before updating its balance.